gno.land/pkg/gnoweb/static/js/renderer.js

```diff
@@ -7,12 +7,87 @@ function renderUsernames(raw) {
   return raw.replace(/( |\n)@([_a-z0-9]{5,16})/, "$1[@$2](/r/demo/users:$2)");
 }
 
+// Create text nodes for the quotes with class names
+const createQuoteNode = (quote) => {
+  const node = document.createElement("a");
+  node.textContent = quote;
+  node.className = "hljs-string";
+  return node;
+};
+
 function parseContent(source, isCode) {
   if (isCode) {
+    // replace &amp; with & in code blocks
+    source = source.replace(/&amp;/g, "&");
+
     const highlightedCode = hljs.highlightAuto(source).value;
+
+    // Split the highlighted code into lines
+    const lines = highlightedCode.split('\n');
+
+    // Add line numbers to each line
+    const numberedLines = lines.map((line, index) => {
+      return `<asp class="number">${index + 1}</asp> ${line}`;
```

Comment:
The added lines get selected when I select code on the page. For instance, copying code from the page yields this:

```
1 package gnoblog
2 
3 import (
4 	"std"
5 
6 	"[gno.land/p/demo/blog](http://localhost:8888/p/demo/blog)"
7 )
8 
9 var b = &blog.Blog{
10 	Title:  "Gnoland's Blog",
11 	Prefix: "/r/gnoland/blog:",
12 }
13 
14 func AddComment(postSlug, comment string) {
15 	assertIsCommenter()
16 	assertNotInPause()
17 
18 	caller := std.GetOrigCaller()
19 	err := b.GetPost(postSlug).AddComment(caller, comment)
20 	checkErr(err)
21 }
22 
23 func Render(path string) string {
24 	return b.Render(path)
25 }
26 
27 func RenderLastPostsWidget(limit int) string {
28 	return b.RenderLastPostsWidget(limit
```

The line numbers shouldn't be selected, similarly to how it happens on GitHub and many other source code displays on the web.

Reply:
https://github.com/gnolang/gno/pull/2757/commits/c79bdc383e8f5a52329fcc409d2000fc32d3fb0b